### PR TITLE
User classes

### PR DESCRIPTION
**Pull request by heqiu**

### CODE REVIEW

File: app/controllers/admin_controller.rb [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    end
+    redirect_to admin_index_path
+  end
+
+  def user_id
+    params.require(:admin_id)
+  end
+
+  def verify_min_admin
+    (!User.find(params.require(:admin_id)).has_role? :admin) || (User.with_role(:admin).length > 1)
```
Comment (brentwalther) [inline app/controllers/admin_controller.rb]: can you replace `params.require` with `user_id` here

File: app/models/ability.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+      can :read, :all
+      can :write, :all
+    elsif user.has_role? :readWrite
+      can :read, :all
+      can :write, :all
+    elsif user.has_role? :readOnly
+      can :read, :all
+    end
+  end
+end
```
Comment (brentwalther) [inline app/models/ability.rb]: Please add a new line at the end of file.

File: config/initializers/rolify.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+Rolify.configure do |config|
+  # By default ORM adapter is ActiveRecord. uncomment to use mongoid
+  # config.use_mongoid
+
+  # Dynamic shortcuts for User class (user.is_admin? like methods). Default is: false
+  # config.use_dynamic_shortcuts
+end
```
Comment (brentwalther) [inline config/initializers/rolify.rb]: newline please

File: config/routes.rb [hunk 4]
```diff
@@ -1,5 +1,7 @@
 Rails.application.routes.draw do
 
+  match "/admin", to: "admin#index", :via => 'get'
```
Comment (brentwalther) [inline config/routes.rb]: this can be changed to `get "/admin", to: "admin#index"`

File: app/controllers/admin_controller.rb [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    end
+  end
+
+  def index
+    @users = User.all
+    render layout: "layouts/application"
+  end
+
+  def change_to_admin
+    User.find(user_id).roles = []
```
Comment (brentwalther) [inline app/controllers/admin_controller.rb]: Instead of assigning an empty array, it would look a bit cleaner to user `.clear`

**Comment by brentwalther** [comment]
This looks pretty good overall. One thing I am a bit worried about is the base that your branch is on. I think it's pretty far behind HEAD. Merging may be very painful.

**Comment by heqiu** [comment]
alright, I'll muck through that tonight haha

**Pull request merged**